upgrading/automated_upgrades.adoc

```diff
@@ -756,6 +756,25 @@ You can set the `openshift_logging_elasticsearch_replace_configmap` parameter to
 ConfigMap can cause the upgrade to fail. The default is set to `false`. For more information, see the parameter 
 in xref:../install_config/aggregate_logging.adoc#aggregate-logging-ansible-variables[specify logging Ansible variables].
 
+. Back up disks.
```

Comment:
@jcantrill What needs to be backed up? Later versions of the ES docs talk about backing up your data using snapshots. Is that what's needed here?

Reply:
@mburke5678  I would itemize this as 'Backup persistent volumes'.  Not sure if we have general PV procedure for doing thath.

@lukas-vlcek we should probably scale down ES prior to upgrade should we not?  This would allow proper backup of the persistentvolumes.  I don't think would advise backup via snapshots unless you know otherwise.

Reply:
@jcantrill First, I would point to this [Elasticsearch Snapshot](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html) documentation page where it highlights the following warning:

> You cannot back up an Elasticsearch cluster by simply taking a copy of the data directories of all of its nodes. Elasticsearch may be making changes to the contents of its data directories while it is running; copying its data directories cannot be expected to capture a consistent picture of their contents. If you try to restore a cluster from such a backup, it may fail and report corruption and/or missing files. Alternatively, it may appear to have succeeded though it silently lost some of its data. The only reliable way to back up a cluster is by using the snapshot and restore functionality.

If we want to do a backup of individual folder on data nodes then keep in mind that node data folders also contain complete cluster state. This means that if you scale down to the point where cluster becomes yellow (it is not possible to allocate replicas) before backup then this is also what user will get when restoring from this backup. Or let's put it this way, if user scales down to 3 nodes then he/she has to restore into three nodes as well. But still this sounds quite fragile to me.

IMO we shall support proper Snapshot and Restore API.

Reply:
What we do in future is not going to be relevant to this documentation. We are strictly writing to address the 2.x to 5.x upgrade.  We addressed this issue when opened by using AWS backups.  I believe we need to advise users to bring the entire system down, snapshot, upgrade recognizing they may lose logs in during upgrade.

Reply:
@jcantrill Ok, in that case I would suggest scaling fluentd pods down first, then calling [Index flush](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/indices-flush.html) on all relevant indices (\*) to make sure all logs are persisted from memory to disk and they do not get lost when ES is shutdown.
(\*) By relevant indices I mean primarily indices holding logs, but I think we can call it on all indices, including SG or Kibana, should there be any update in the flux...

Reply:
@lukas-vlcek Added a step between scaling down and snapshotting:

_Perform an Elasticsearch Index flush on all relevant indices. The flush process persists all logs from memory to disk, which prevents log loss when Elasticsearch is shutdown during the upgrade._

Reply:
The user should scale down the fluentd pods? @jcantrill suggests the user `scale down ES`.

Reply:
scaling down fluent will stop data ingestion but scaling down ES will also ensure the cluster state is not changing so you could say snapshot the AWS volumes